client/landing/stepper/declarative-flow/transferring-hosted-site-flow.ts

```diff
@@ -29,10 +30,15 @@ const transferringHostedSite: Flow = {
 	useStepNavigation( currentStep, navigate ) {
 		const flowName = this.name;
 		const siteId = useSiteIdParam();
+		const exitTo = useQuery().get( 'to' ) || 'home';
 		const dispatch = useReduxDispatch();
 
 		const exitFlow = ( to: string ) => {
-			window.location.assign( to );
+			const timeoutDuration = to === 'home' ? 0 : 3000;
+
+			setTimeout( () => {
+				window.location.assign( to );
+			}, timeoutDuration );
```

Comment:
After the site Atomic transfer is complete, we still need a few moments for "things" to update. Redirecting right away does not work.

I tried to avoid this `setTimeout` by leveraging `requestSite( siteId )` and trying waiting on various "promises" and "states", but ultimately I was only able to get it working consistently with this timeout.

I set a conditional on `timeoutDuration` so that we don't add unnecessary process time to the original flow. It only waits for our flow as it is.

I believe this method to be adequate, but welcome better alternatives!

Reply:
Could the [WatiForAtomic](https://github.com/Automattic/wp-calypso/blob/e7a56a71894010c157b51bc8f038acbab3ee62d8/client/landing/stepper/declarative-flow/internals/steps-repository/wait-for-atomic/index.tsx#L36) support the `feature` query string?

If it's provided, the step will poll the latest site data and do the redirection only when the feature is ready. What do you think?

Reply:
Done by https://github.com/Automattic/wp-calypso/pull/91141/commits/3886a3421bd2a585c59f84a5f2d5dd519c429e2d